partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -289,5 +297,99 @@ public Long extract(ConnectRecord<?> record) {
         throw new PartitionException("Error encoding partition.");
       }
     }
+
+    private Object getField(Struct struct, String fieldName) {
+      Object field;
+      try {
+        field = struct.get(fieldName);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The field named '%s' does not exist.", fieldName), e);
+      }
+      return field;
+    }
+
+    private Object getNestedFieldValue(Struct struct) {
+      final String[] fieldNames = fieldName.split("\\.");
+      Struct tmpStruct = struct;
+      Object tmpObject;
+      try {
+        // Iterate down to final struct
+        int i = 0;
+        while (i < fieldNames.length - 1) {
```

Comment:
since you're using Java, I think you can just iterate over the references using:

```
for (name :fieldNames) {
    #name is a read only iterable
}
```
This ensures cleaner, read-only iteration, good practice.

Reply:
I tried with for loop, the code might be little bit easier to understand, but still complex due to the fact that it/s nested format, and the way the connect struct works.

Reply:
Your loop is also functionally equivalent to:
for(int i = 0; i < fieldNames.length - 1; i++) { }

I would insist that you do not use a while-loop for an integer-based count down. It is not good practice.